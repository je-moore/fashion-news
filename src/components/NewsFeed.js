import React from 'react'
import NewsCard from './NewsCard'
import { useNewsFeedStyles } from '../styles'

function NewsFeed({ data }) {
  const classes = useNewsFeedStyles()

  return (
    <div className={classes.cardContainer}>
      {data.fashionunitedNlNewsArticles.map(article => (
        <NewsCard article={article} key={article.id} />
      ))}
    </div>
  )
}

export default NewsFeed
