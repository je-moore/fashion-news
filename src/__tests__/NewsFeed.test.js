import React from 'react'
import EnzymeToJson from 'enzyme-to-json'
import { mount } from 'enzyme'
import NewsFeed from '../components/NewsFeed'
import { useHistory } from 'react-router-dom'

jest.mock('react-router-dom', () => {
  return {
    __esModule: true,
    useParams: jest.fn(() => {
      return {
        modalId: 44366,
      }
    }),
    useHistory: jest.fn(() => {
      return {
        length: 50,
        action: 'PUSH',
        location: {
          pathname: '/',
          search: '',
          hash: '',
          key: 'ud1unh',
        },
      }
    }),
    Link: jest.fn(() => null),
  }
})

const data = {
  fashionunitedNlNewsArticles: [
    {
      title:
        'Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.',
      url:
        'https://fashionunited.nl/nieuws/mode/interview-martijn-vlaming-van-oger-over-zijn-rol-als-personal-advisor-bij-italiaans-luxury-maatwerk/2019121344367',
      imageUrl:
        '/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn',
      description:
        'Een carriÃ¨re in de modebranche is een belangrijk onderwerp op FashionUnited. Met deze interviewserie',
      id: 44367,
      __typename: 'LocalNewsArticle',
    },
    {
      title:
        'Wolford maakt nieuw plan voor herpositionering merk na omzetdaling en operationeel verlies',
      url:
        'https://fashionunited.nl/nieuws/business/wolford-maakt-nieuw-plan-voor-herpositionering-merk-na-omzetdaling-en-operationeel-verlies/2019121344366',
      imageUrl: null,
      description:
        'Wolford AG zag de omzet in het het eerste halfjaar met 3 procent dalen, zo blijkt uit het halfjaar verslag. Het operationele verlies komt neer op 9,4 miljoen ',
      id: 44366,
      __typename: 'LocalNewsArticle',
    },
  ],
}

const fetchMore = () => null

it('renders correctly', () => {
  const subject = mount(<NewsFeed data={data} fetchMore={fetchMore} />)
  expect(EnzymeToJson(subject)).toMatchSnapshot()
})
