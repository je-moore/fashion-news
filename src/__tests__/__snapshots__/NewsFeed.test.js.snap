// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<NewsFeed
  data={
    Object {
      "fashionunitedNlNewsArticles": Array [
        Object {
          "__typename": "LocalNewsArticle",
          "description": "Een carrière in de modebranche is een belangrijk onderwerp op FashionUnited. Met deze interviewserie",
          "id": 44367,
          "imageUrl": "/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn",
          "title": "Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.",
          "url": "https://fashionunited.nl/nieuws/mode/interview-martijn-vlaming-van-oger-over-zijn-rol-als-personal-advisor-bij-italiaans-luxury-maatwerk/2019121344367",
        },
        Object {
          "__typename": "LocalNewsArticle",
          "description": "Wolford AG zag de omzet in het het eerste halfjaar met 3 procent dalen, zo blijkt uit het halfjaar verslag. Het operationele verlies komt neer op 9,4 miljoen ",
          "id": 44366,
          "imageUrl": null,
          "title": "Wolford maakt nieuw plan voor herpositionering merk na omzetdaling en operationeel verlies",
          "url": "https://fashionunited.nl/nieuws/business/wolford-maakt-nieuw-plan-voor-herpositionering-merk-na-omzetdaling-en-operationeel-verlies/2019121344366",
        },
      ],
    }
  }
  fetchMore={[Function]}
>
  <div
    className="makeStyles-cardContainer-1"
  >
    <NewsCard
      article={
        Object {
          "__typename": "LocalNewsArticle",
          "description": "Een carrière in de modebranche is een belangrijk onderwerp op FashionUnited. Met deze interviewserie",
          "id": 44367,
          "imageUrl": "/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn",
          "title": "Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.",
          "url": "https://fashionunited.nl/nieuws/mode/interview-martijn-vlaming-van-oger-over-zijn-rol-als-personal-advisor-bij-italiaans-luxury-maatwerk/2019121344367",
        }
      }
      key="44367"
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-card-2"
        onClick={[Function]}
      >
        <ForwardRef(Card)
          className="makeStyles-card-2"
          classes={
            Object {
              "root": "MuiCard-root",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Paper))
            className="MuiCard-root makeStyles-card-2"
            elevation={1}
            onClick={[Function]}
          >
            <ForwardRef(Paper)
              className="MuiCard-root makeStyles-card-2"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0",
                  "elevation1": "MuiPaper-elevation1",
                  "elevation10": "MuiPaper-elevation10",
                  "elevation11": "MuiPaper-elevation11",
                  "elevation12": "MuiPaper-elevation12",
                  "elevation13": "MuiPaper-elevation13",
                  "elevation14": "MuiPaper-elevation14",
                  "elevation15": "MuiPaper-elevation15",
                  "elevation16": "MuiPaper-elevation16",
                  "elevation17": "MuiPaper-elevation17",
                  "elevation18": "MuiPaper-elevation18",
                  "elevation19": "MuiPaper-elevation19",
                  "elevation2": "MuiPaper-elevation2",
                  "elevation20": "MuiPaper-elevation20",
                  "elevation21": "MuiPaper-elevation21",
                  "elevation22": "MuiPaper-elevation22",
                  "elevation23": "MuiPaper-elevation23",
                  "elevation24": "MuiPaper-elevation24",
                  "elevation3": "MuiPaper-elevation3",
                  "elevation4": "MuiPaper-elevation4",
                  "elevation5": "MuiPaper-elevation5",
                  "elevation6": "MuiPaper-elevation6",
                  "elevation7": "MuiPaper-elevation7",
                  "elevation8": "MuiPaper-elevation8",
                  "elevation9": "MuiPaper-elevation9",
                  "root": "MuiPaper-root",
                  "rounded": "MuiPaper-rounded",
                }
              }
              elevation={1}
              onClick={[Function]}
            >
              <div
                className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-2 MuiPaper-rounded"
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(CardContent))>
                  <ForwardRef(CardContent)
                    classes={
                      Object {
                        "root": "MuiCardContent-root",
                      }
                    }
                  >
                    <div
                      className="MuiCardContent-root"
                    >
                      <ArticleImage
                        imageUrl="/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn"
                      >
                        <img
                          alt=""
                          className="makeStyles-articleImg-33"
                          src="https://fashionunited.com/DlQNIXLzw12YzrwK9k3UutFViomv5EVKEyGRmKDAsTw/gravity:sm/quality:70/aHR0cHM6Ly9zdGF0aWMuZmFzaGlvbnVuaXRlZC5jb20vMjAxOTEyLzRzcXVhcmUtbmluZTEuanBn"
                        />
                      </ArticleImage>
                      <WithStyles(ForwardRef(Typography))
                        className="makeStyles-title-3"
                        color="textPrimary"
                        gutterBottom={true}
                        variant="h4"
                      >
                        <ForwardRef(Typography)
                          className="makeStyles-title-3"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          color="textPrimary"
                          gutterBottom={true}
                          variant="h4"
                        >
                          <h4
                            className="MuiTypography-root makeStyles-title-3 MuiTypography-h4 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
                          >
                            Interview: Martijn Vlaming van OGER over zijn rol als Personal Advisor bij Italiaans Luxury Maatwerk.
                          </h4>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </div>
                  </ForwardRef(CardContent)>
                </WithStyles(ForwardRef(CardContent))>
              </div>
            </ForwardRef(Paper)>
          </WithStyles(ForwardRef(Paper))>
        </ForwardRef(Card)>
      </WithStyles(ForwardRef(Card))>
    </NewsCard>
    <NewsCard
      article={
        Object {
          "__typename": "LocalNewsArticle",
          "description": "Wolford AG zag de omzet in het het eerste halfjaar met 3 procent dalen, zo blijkt uit het halfjaar verslag. Het operationele verlies komt neer op 9,4 miljoen ",
          "id": 44366,
          "imageUrl": null,
          "title": "Wolford maakt nieuw plan voor herpositionering merk na omzetdaling en operationeel verlies",
          "url": "https://fashionunited.nl/nieuws/business/wolford-maakt-nieuw-plan-voor-herpositionering-merk-na-omzetdaling-en-operationeel-verlies/2019121344366",
        }
      }
      key="44366"
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-card-2"
        onClick={[Function]}
      >
        <ForwardRef(Card)
          className="makeStyles-card-2"
          classes={
            Object {
              "root": "MuiCard-root",
            }
          }
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Paper))
            className="MuiCard-root makeStyles-card-2"
            elevation={1}
            onClick={[Function]}
          >
            <ForwardRef(Paper)
              className="MuiCard-root makeStyles-card-2"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0",
                  "elevation1": "MuiPaper-elevation1",
                  "elevation10": "MuiPaper-elevation10",
                  "elevation11": "MuiPaper-elevation11",
                  "elevation12": "MuiPaper-elevation12",
                  "elevation13": "MuiPaper-elevation13",
                  "elevation14": "MuiPaper-elevation14",
                  "elevation15": "MuiPaper-elevation15",
                  "elevation16": "MuiPaper-elevation16",
                  "elevation17": "MuiPaper-elevation17",
                  "elevation18": "MuiPaper-elevation18",
                  "elevation19": "MuiPaper-elevation19",
                  "elevation2": "MuiPaper-elevation2",
                  "elevation20": "MuiPaper-elevation20",
                  "elevation21": "MuiPaper-elevation21",
                  "elevation22": "MuiPaper-elevation22",
                  "elevation23": "MuiPaper-elevation23",
                  "elevation24": "MuiPaper-elevation24",
                  "elevation3": "MuiPaper-elevation3",
                  "elevation4": "MuiPaper-elevation4",
                  "elevation5": "MuiPaper-elevation5",
                  "elevation6": "MuiPaper-elevation6",
                  "elevation7": "MuiPaper-elevation7",
                  "elevation8": "MuiPaper-elevation8",
                  "elevation9": "MuiPaper-elevation9",
                  "root": "MuiPaper-root",
                  "rounded": "MuiPaper-rounded",
                }
              }
              elevation={1}
              onClick={[Function]}
            >
              <div
                className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-2 MuiPaper-rounded"
                onClick={[Function]}
              >
                <WithStyles(ForwardRef(CardContent))>
                  <ForwardRef(CardContent)
                    classes={
                      Object {
                        "root": "MuiCardContent-root",
                      }
                    }
                  >
                    <div
                      className="MuiCardContent-root"
                    >
                      <ArticleImage
                        imageUrl={null}
                      >
                        <img
                          alt=""
                          className="makeStyles-articleImg-33 makeStyles-defaultImg-34"
                          src="https://fashionunited.info/global-assets/img/default/fu-default_1200x630_black-favicon.jpg"
                        />
                      </ArticleImage>
                      <WithStyles(ForwardRef(Typography))
                        className="makeStyles-title-3"
                        color="textPrimary"
                        gutterBottom={true}
                        variant="h4"
                      >
                        <ForwardRef(Typography)
                          className="makeStyles-title-3"
                          classes={
                            Object {
                              "alignCenter": "MuiTypography-alignCenter",
                              "alignJustify": "MuiTypography-alignJustify",
                              "alignLeft": "MuiTypography-alignLeft",
                              "alignRight": "MuiTypography-alignRight",
                              "body1": "MuiTypography-body1",
                              "body2": "MuiTypography-body2",
                              "button": "MuiTypography-button",
                              "caption": "MuiTypography-caption",
                              "colorError": "MuiTypography-colorError",
                              "colorInherit": "MuiTypography-colorInherit",
                              "colorPrimary": "MuiTypography-colorPrimary",
                              "colorSecondary": "MuiTypography-colorSecondary",
                              "colorTextPrimary": "MuiTypography-colorTextPrimary",
                              "colorTextSecondary": "MuiTypography-colorTextSecondary",
                              "displayBlock": "MuiTypography-displayBlock",
                              "displayInline": "MuiTypography-displayInline",
                              "gutterBottom": "MuiTypography-gutterBottom",
                              "h1": "MuiTypography-h1",
                              "h2": "MuiTypography-h2",
                              "h3": "MuiTypography-h3",
                              "h4": "MuiTypography-h4",
                              "h5": "MuiTypography-h5",
                              "h6": "MuiTypography-h6",
                              "noWrap": "MuiTypography-noWrap",
                              "overline": "MuiTypography-overline",
                              "paragraph": "MuiTypography-paragraph",
                              "root": "MuiTypography-root",
                              "srOnly": "MuiTypography-srOnly",
                              "subtitle1": "MuiTypography-subtitle1",
                              "subtitle2": "MuiTypography-subtitle2",
                            }
                          }
                          color="textPrimary"
                          gutterBottom={true}
                          variant="h4"
                        >
                          <h4
                            className="MuiTypography-root makeStyles-title-3 MuiTypography-h4 MuiTypography-colorTextPrimary MuiTypography-gutterBottom"
                          >
                            Wolford maakt nieuw plan voor herpositionering merk na omzetdaling en operationeel verlies
                          </h4>
                        </ForwardRef(Typography)>
                      </WithStyles(ForwardRef(Typography))>
                    </div>
                  </ForwardRef(CardContent)>
                </WithStyles(ForwardRef(CardContent))>
              </div>
            </ForwardRef(Paper)>
          </WithStyles(ForwardRef(Paper))>
        </ForwardRef(Card)>
      </WithStyles(ForwardRef(Card))>
    </NewsCard>
  </div>
  <LoadMoreButton
    fetchMore={[Function]}
    offset={2}
  >
    <div
      className="makeStyles-loadMore-65"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        <ForwardRef(Button)
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit",
              "contained": "MuiButton-contained",
              "containedPrimary": "MuiButton-containedPrimary",
              "containedSecondary": "MuiButton-containedSecondary",
              "containedSizeLarge": "MuiButton-containedSizeLarge",
              "containedSizeSmall": "MuiButton-containedSizeSmall",
              "disableElevation": "MuiButton-disableElevation",
              "disabled": "Mui-disabled",
              "endIcon": "MuiButton-endIcon",
              "focusVisible": "Mui-focusVisible",
              "fullWidth": "MuiButton-fullWidth",
              "iconSizeLarge": "MuiButton-iconSizeLarge",
              "iconSizeMedium": "MuiButton-iconSizeMedium",
              "iconSizeSmall": "MuiButton-iconSizeSmall",
              "label": "MuiButton-label",
              "outlined": "MuiButton-outlined",
              "outlinedPrimary": "MuiButton-outlinedPrimary",
              "outlinedSecondary": "MuiButton-outlinedSecondary",
              "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
              "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
              "root": "MuiButton-root",
              "sizeLarge": "MuiButton-sizeLarge",
              "sizeSmall": "MuiButton-sizeSmall",
              "startIcon": "MuiButton-startIcon",
              "text": "MuiButton-text",
              "textPrimary": "MuiButton-textPrimary",
              "textSecondary": "MuiButton-textSecondary",
              "textSizeLarge": "MuiButton-textSizeLarge",
              "textSizeSmall": "MuiButton-textSizeSmall",
            }
          }
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          <WithStyles(ForwardRef(ButtonBase))
            className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            type="button"
          >
            <ForwardRef(ButtonBase)
              className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              component="button"
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="Mui-focusVisible"
              onClick={[Function]}
              type="button"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Load More
                </span>
                <NoSsr>
                  <WithStyles(undefined)
                    center={false}
                  >
                    <ForwardRef(TouchRipple)
                      center={false}
                      classes={
                        Object {
                          "child": "MuiTouchRipple-child",
                          "childLeaving": "MuiTouchRipple-childLeaving",
                          "childPulsate": "MuiTouchRipple-childPulsate",
                          "ripple": "MuiTouchRipple-ripple",
                          "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                          "rippleVisible": "MuiTouchRipple-rippleVisible",
                          "root": "MuiTouchRipple-root",
                        }
                      }
                    >
                      <span
                        className="MuiTouchRipple-root"
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          component={null}
                          exit={true}
                        />
                      </span>
                    </ForwardRef(TouchRipple)>
                  </WithStyles(undefined)>
                </NoSsr>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(Button)>
      </WithStyles(ForwardRef(Button))>
    </div>
  </LoadMoreButton>
</NewsFeed>
`;
